### PR TITLE
[feat] Connection Microsoft

### PR DESCRIPTION
**Pull request by Tristancalard**

### CODE REVIEW

File: backend/src/app.ts [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+      httpOnly: true,
+      secure: false,
+    },
+  })
+);
+
+app.use(logger('dev'));
```
Comment (thmarinho) [inline backend/src/app.ts]: Should adapt to each deployment stage (dev, preprod, prod etc.)

File: backend/src/routes/users/users.query.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * @param {string} accessToken - Raw access token string
+ * @returns {Promise<any>} - Result of the request
+ */
+async function user_connection(endpoint: string, accessToken: string): Promise<any> {
+    const options = {
+        headers: {
+            Authorization: `Bearer ${accessToken}`,
+        },
+    };
+    console.log(`Requête effectuée à ${endpoint} à : ` + new Date().toString());
```
Comment (thmarinho) [inline backend/src/routes/users/users.query.ts]: Should be in english. Moreover, that's not necessary, you're using a logger above

File: backend/src/routes/users/users.query.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    try {
+        const response: AxiosResponse = await axios.get(endpoint, options);
+        return response.data;
```
Comment (thmarinho) [inline backend/src/routes/users/users.query.ts]: make it using fetch, axios will only make the app bigger

File: backend/src/routes/users/users.ts
Comment (thmarinho) [inline backend/src/routes/users/users.ts:1]: Rename this file, we should know it's the router

File: backend/src/routes/users/users.query.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+import axios, { AxiosResponse } from 'axios';
+
+/**
+ * Attaches a given access token to an MS Graph API call
+ * @param {string} endpoint - REST API endpoint to call
+ * @param {string} accessToken - Raw access token string
+ * @returns {Promise<any>} - Result of the request
+ */
```
Comment (thmarinho) [inline backend/src/routes/users/users.query.ts]: Remove this kind of comments, you'll propably use another documentation manager

File: backend/src/routes/users/users.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+function isAuthenticated(req: Request, res: Response, next: NextFunction): void {
+    if (!req.session.isAuthenticated) {
+        return res.redirect('/auth/signin'); // Redirect to the sign-in route
+    }
+    next();
+}
+
+router.get('/id', isAuthenticated, async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    res.render('id', { idTokenClaims: req.session.account.idTokenClaims });
+});
```
Comment (thmarinho) [inline backend/src/routes/users/users.ts]: Why is there some `res.render` ? You're backend should not send views

**Review by thmarinho — CHANGES_REQUESTED** [review]
Have this routes been tested ? You should be able to get an Office token allowing to get log on the intranet for example

File: backend/src/app.ts [hunk 6]
```diff
@@ -3,12 +3,14 @@ import session from 'express-session';
...
 import logger from 'morgan';
 
-import usersRouter from './routes/users/users.js';
+import usersRouter from './routes/users/router_users.js';
 import authRouter from './routes/auth/auth.js';
 
 const app = express();
 const port = 3000;
 
+const logger_mode = 1;
```
Comment (alanschnegg) [inline backend/src/app.ts:12]: Should be in .env

File: backend/src/routes/users/users.query.ts [hunk 7]
```diff
@@ -5,19 +5,13 @@
...
 async function user_connection(endpoint: string, accessToken: string): Promise<any> {
     const options = {
         headers: {
             Authorization: `Bearer ${accessToken}`,
         },
     };
-    console.log(`Requête effectuée à ${endpoint} à : ` + new Date().toString());
+    console.log(`Request made to ${endpoint} at : ` + new Date().toString());
     try {
         const response: AxiosResponse = await axios.get(endpoint, options);
```
Comment (alanschnegg) [inline backend/src/routes/users/users.query.ts]: Replace Axios with fetch

**Review by alanschnegg — CHANGES_REQUESTED** [review]

File: backend/src/app.ts [hunk 8]
```diff
@@ -1,13 +1,40 @@
...
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+      httpOnly: true,
+      secure: false,
+    },
+  })
+);
+
+app.use(logger(logger_mode ? 'dev' : 'common'));
```
Comment (thmarinho) [inline backend/src/app.ts]: `process.env.NODE_ENV === 'prod' ? 'dev : 'common'`

**Review by thmarinho — CHANGES_REQUESTED** [review]

**Comment by Tristancalard** [comment]
It's done :+1:

**Comment by thmarinho** [comment]
Remove the comments from the Microsoft files, the code is explicit, you don't need to say 'fetching this route' on a request

**Review by thmarinho — CHANGES_REQUESTED** [review]
Remove the comments from the Microsoft files, the code is explicit, you don't need to say 'fetching this route' on a request

File: backend/src/routes/users/users.query.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+ * Licensed under the MIT License.
+ */
+
+async function user_connection(endpoint: string, accessToken: string): Promise<any> {
+    const options = {
+        headers: {
+            Authorization: `Bearer ${accessToken}`,
+        },
+    };
+    console.log(`Request made to ${endpoint} at : ` + new Date().toString());
```
Comment (thmarinho) [inline backend/src/routes/users/users.query.ts]: This line should still be deleted

File: backend/src/app.ts [hunk 10]
```diff
@@ -1,13 +1,41 @@
...
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+      httpOnly: true,
+      secure: false,
+    },
+  })
+);
+
+app.use(logger(process.env.NODE_ENV === 'prod' ? 'dev' : 'common'));
```
Comment (thmarinho) [inline backend/src/app.ts]:
```js
app.use(process.env.NODE8ENV === 'prod' ? 'common' : 'dev')
```

**Review by thmarinho — CHANGES_REQUESTED** [review]

**Review by alanschnegg — APPROVED** [review]

**Review by thmarinho — APPROVED** [review]

**Pull request merged**